mixer/cmd/root.go

```diff
@@ -62,6 +63,39 @@ var RootCmd = &cobra.Command{
 				os.Exit(0)
 			}
 		}
+
+		// if '--native', check if format missmatch
+		if builder.Native {
+			var version string
+			if cmd.Name() == "init" {
+				version = initFlags.clearVer
+			}
+			b := builder.New()
+			hostFormat, upstreamFormat, err := b.GetHostAndUpstreamFormats(version)
+			if err != nil {
+				return err
+			}
+
+			if hostFormat == "" {
+				fmt.Println("Warning: Unable to determine host format. Running natively may fail.")
+			} else if hostFormat != upstreamFormat {
+				fmt.Println("Warning: The host format and mix upstream format do not match.",
+					"Mixer may be incompatible with this format; running natively may fail.")
+			}
+		} else if cmd.Name() != "init" { // Init needs to be handled differently because there is no config yet
+			b, err := builder.NewFromConfig(config)
+			if err != nil {
+				fail(err)
+			}
+			if err := b.RunCommandInContainer(reconstructCommand(cmd, args)); err != nil {
+				fail(err)
+			}
+			// Cancel native run
+			cmd.RunE = nil
+			cmd.Run = func(cmd *cobra.Command, args []string) {} // No-op
```

Comment:
😃

Reply:
They so far haven't responded to my PR (https://github.com/spf13/cobra/pull/657) to make this noop hack unnecessary.